src/realm/obj.cpp

```diff
@@ -2325,9 +2327,19 @@ void Obj::assign_pk_and_backlinks(const Obj& other)
             // Object has been deleted - we are done
             return IteratorControl::Stop;
         }
+
         auto t = m_table->get_opposite_table(col);
         auto c = m_table->get_opposite_column(col);
         auto backlinks = other.get_all_backlinks(col);
+
```

Comment:
Clever idea. It leaks the implementation of backlinks storage a bit here, but we have accepted worse in the name of performance.